sealing/enclave/src/lib.rs

```diff
@@ -0,0 +1,65 @@
+#![no_std]
+#![deny(unsafe_op_in_unsafe_fn)]
+
+use std::{eprintln, slice};
+
+use sealing::{Nonce, SecretKey};
+use sgx_keys::{SgxKeyPolicy, SgxSecretBuilder};
+use sgx_rand::{Rng, SgxRng};
+use sgx_tstd as std;
+use sgx_types::sgx_status_t;
+
+/// Seal private user data.
+#[no_mangle]
+pub unsafe extern "C" fn seal_user_data(
+    // input parameters
+    user_id: *const u8,
+    user_id_length: usize,
+    // dual-purpose parameters
+    user_data: *mut u8,
+    // output parameters
+    sealed_user_data_size: usize,
+    sealed_user_sealing_key: *mut u8,
+) -> sgx_status_t {
+    let user_id = unsafe { slice::from_raw_parts(user_id, user_id_length) };
+    let user_data_buf = unsafe { slice::from_raw_parts(user_data, sealed_user_data_size) };
+
+    let mut rng = SgxRng::new().expect("SGX: RDRAND instruction failed");
+    let mut user_sealing_key = SecretKey::new([0u8; 32]);
```

Comment:
`32` feels like it should an associated type of SecretKey, such that...
```rust
SecretKey::new([0u8; 32])
```
Maybe even better is to do a Default derive on SecretKey, so that you could just...
```rust
SecretKey::default()
```